### DISCUSSION TITLE
Add documentation on third-party modules

### DISCUSSION THREAD

**Discussion by addisonamiri** [zimfw/zimfw#10] [Ideas]
Some documentation on the recommended way to use third-party modules would be nice to have in the wiki. There are some prezto modules I'd like to use until they get ported to zim (if that's even on the roadmap).

**Comment by Eriner** [comment]
Did you check the custom folder? That's what it's for. Alternatively, you should be able to (for the most part) copy modules from prezto into zim. Though, they may contain prezto-specific things that will break functionality. What would you like ported over, specifically?

**Comment by addisonamiri** [comment]
`editor` and `fasd` are the only two I was using and aren't in zim yet. I'll throw them in a custom folder for now. What I meant by more documentation was something closer to how antigen specifies how to use other plugins with its framework.

**Comment by Eriner** [comment]
Using other plugins (copy-pasta from prezto/oh-my-zsh) may not work in some situations, just so you're aware. I have zero interest in supporting officially using modules/plugins from other repos (like how antigen works). If it's worthwhile, I'll add it as an official module; anything else is up to the user to add and maintain. 
Make sure you `source fasd/fasd.zsh` to the `init.zsh` in the custom folder, otherwise they won't be loaded.

I'll look into adding a `fasd` module. What, specifically, did you want from the `editor` module? There is an `input` module that handles very basic (usability) bindkeys, but nothing else.

**Comment by addisonamiri** [comment]
I just used the editor module for basic vim-esque editing support. I'm a
bit OCD and if there's a module for it I rather use that instead of putting
more stuff in my zshrc :P

The fasd module was pretty useful but I've used other utilities that do the
same thing. It's basically just remembering and sorting visited directories
so you can type `j sillyfolder` and it would cd into
`~/one/directory/two/directory/red/directory/blue/directory/sillyfolder`
for you. I'll throw it in my custom folder for now but if you end up
supporting it that'd be cool.

**Comment by Eriner** [comment]
Yeah, seeing as there are so many of them (`fasd`, `z`, `enhancd` etc) I probably won't add _any_ of them. I will leave this choice up to each individual user, if they chose to use it. I find those types of 'enhancements' to be bothersome. I would rather just use tab completion and type. Having to constantly guess about what it may or may not be 'smart' enough to understand would add irritation to my workflow.

So, if you would like those things, I suggest adding them to your custom folder. I will work adding more ZLE options (vi/emacs choices, etc) over the next week. Until then, I will leave this issue open to remind myself.

**Comment by addisonamiri** [comment]
Haha yeah that's fine. I look forward to your future work ^_^

**Comment by Eriner** [comment]
> I will work adding more ZLE options (vi/emacs choices, etc) over the next week.

I've decided against this. Zsh should pick up the correct bindkeys from `${EDITOR}`. Unless someone explicitly requests this, I don't see a reason to add it. 

I'm going to close this issue. Like I've mentioned before, if there is anything specific that you want, let me know. 

Thanks for your feedback!
